pkg/check/datastore.go

```diff
@@ -150,9 +157,10 @@ func CheckDefaultDatastore(ctx *CheckContext) error {
 
 func checkDefaultDatastoreWithDSType(ctx *CheckContext, dsTypes dataStoreTypeCollector) error {
 	for _, fd := range ctx.PlatformSpec.FailureDomains {
+		vCenter := ctx.VCenters[fd.Server]
```

Comment:
can `VCenters` miss a server? Judging by checks in most other places it can.

Reply:
Normally not, but if there is a mismatch in configs, we can hit this.  I hit it in my testing so I just wanted to make checks for each of these.  I'll make sure this is covered here too.